### PR TITLE
Implement Jest testing platform for all exercises (Solutions branch)

### PR DESCRIPTION
**Pull request by xandora**
This PR overhauls the JavaScript exercises to use Jest instead of Jasmine. This is the Solutions branch version, and should be identical except for the implemented solutions.

- Updated README.md content.
- Updated syntax to be consistent across exercises.
- Users can simply `npm install` to pull required node modules in using `package.json` file.
- ESLint has been included, but is set to run manually as a forward thinking possibility for a "Linting lesson" using existing lessons.

### CODE REVIEW

File: helloWorld/helloWorld.spec.js [hunk 1]
```diff
@@ -1,7 +1,5 @@
 const helloWorld = require('./helloWorld');
 
-describe('Hello World', function() {
-  it('says hello world', function() {
-    expect(helloWorld()).toEqual('Hello, World!');
-  });
-});
+test('says "Hello, World!"', function() {
+  expect(helloWorld()).toBe("Hello, World!");
+});
```
Comment (twalton83) [inline helloWorld/helloWorld.spec.js:5]: Can we have a new line?

File: pigLatin/pigLatin.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  
+const firstVowelIndex = function(string) {
+	const vowels = string.match(/[aeiou]/g);
+	if (vowels[0] == "u" && string[string.indexOf(vowels[0]) - 1] == "q") {
+	  return string.indexOf(vowels[1]);
+	}
+	return string.indexOf(vowels[0]);
+  };
+  
+  module.exports = pigLatin;
```
Comment (twalton83) [inline pigLatin/pigLatin.js:21]: Another new line.

**Review by twalton83 — CHANGES_REQUESTED** [review]
Just a couple small things!

File: helloWorld/helloWorld.spec.js (hunk 1 above)
Comment (xandora) [reply helloWorld/helloWorld.spec.js]: Done.

File: pigLatin/pigLatin.js (hunk 2 above)
Comment (xandora) [reply pigLatin/pigLatin.js]: Done.

**Comment by twalton83** [comment]
Closing because this will be addressed with #150 instead

**Pull request closed without merging**